From 427393f8dc5779cb9ee93719f42c9f5ed1b72ed9 Mon Sep 17 00:00:00 2001
From: Qi Zhang <qi1.zhang@intel.com>
Date: Wed, 10 May 2023 18:05:34 +0800
Subject: [PATCH] alarmtimer: don't set RTC during suspend

The RTC wake up is not used on IVI platform.
It doesn't need alarm timer to wake up system.

Tracked-On: OAM-110029
Signed-off-by: Qi Zhang <qi1.zhang@intel.com>
Signed-off-by: Mallikarjun Chegaraddi, Raju <raju.mallikarjun.chegaraddi@intel.com>
---
 kernel/time/alarmtimer.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/kernel/time/alarmtimer.c b/kernel/time/alarmtimer.c
index 5897828b9d7e..639fb00624e2 100644
--- a/kernel/time/alarmtimer.c
+++ b/kernel/time/alarmtimer.c
@@ -888,7 +888,15 @@ static const struct dev_pm_ops alarmtimer_pm_ops = {
 static struct platform_driver alarmtimer_driver = {
 	.driver = {
 		.name = "alarmtimer",
+		/**
+		* skip alarm timer suspend/resume flow for Android
+		*  auto usage without ENABLE_ALARM_PM_OPS.
+		* we don't rely on RTC. We can set the timer to MCU
+		*  and MCU will wake the SoC
+		*/
+		#if defined(ENABLE_ALARM_PM_OPS)
 		.pm = &alarmtimer_pm_ops,
+		#endif /* ENABLE_ALARM_PM_OPS */
 	}
 };
 
-- 
2.34.1

